modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java

```diff
@@ -0,0 +1,506 @@
+                        getCode().equals(String.valueOf(OperationCode.Command.DEVICE_LOCK))) {
+                    operation.setStatus(Operation.Status.ERROR);
+                    updateOperations(syncmlDocument.getHeader().getSource().getLocURI(), pendingDataOperations);
+                    try {
+                        NotificationManagementService service =
+                                WindowsAPIUtils.getNotificationManagementService();
+                        Notification lockResetNotification = new Notification();
+                        lockResetNotification.setOperationId(status.getCommandReference());
+                        lockResetNotification.setStatus(String.valueOf(Notification.Status.NEW));
+                        lockResetNotification.setDeviceIdentifier(deviceIdentifier);
+                        lockResetNotification.setDescription(
+                                Constants.SyncMLResponseCodes.LOCKRESET_NOTIFICATION);
+                        service.addNotification(lockResetNotification);
+                    } catch (NotificationManagementException e) {
+                        throw new WindowsOperationException("Failure occurred in getting notification service", e);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Update operation statuses
+     *
+     * @param deviceId   specific device Id
+     * @param operations operation list to be update
+     * @throws OperationManagementException
+     */
+    public void updateOperations(String deviceId,
+                                 List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> operations)
+            throws OperationManagementException {
+
+        for (Operation operation : operations) {
+            WindowsAPIUtils.updateOperation(deviceId, operation);
+            if (log.isDebugEnabled()) {
+                log.debug("Updating operation '" + operation.toString() + "'");
+            }
+        }
+    }
+
+    /**
+     * Update Status of the lock operation.
+     *
+     * @param status           Status of the operation.
+     * @param syncmlDocument   parsed syncml payload.
+     * @param deviceIdentifier Device Id.
+     * @throws OperationManagementException
+     * @throws DeviceManagementException
+     * @throws NotificationManagementException
+     */
+    public void lockOperationUpdate(Status status, SyncmlDocument syncmlDocument, DeviceIdentifier deviceIdentifier)
+            throws OperationManagementException, DeviceManagementException, NotificationManagementException {
+
+        pendingDataOperations = SyncmlUtils.getDeviceManagementService()
+                .getOperationsByDeviceAndStatus(deviceIdentifier, Operation.Status.PENDING);
+        if (status.getData().equals(Constants.SyncMLResponseCodes.ACCEPTED)) {
+            for (Operation operation : pendingDataOperations) {
+                if (operation.getCode().equals(OperationCode.Command.DEVICE_LOCK.getCode())
```

Comment:
As a best practice, always compare the non-deterministic variable values against known constants.